extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectionImpl.java

```diff
@@ -211,42 +240,38 @@ public BroadcastSender filter(Predicate<WebSocketConnection> predicate) {
 
         @Override
         public Uni<Void> sendText(String message) {
-            return doSend(new Function<WebSocketConnection, Uni<Void>>() {
-
-                @Override
-                public Uni<Void> apply(WebSocketConnection c) {
-                    return c.sendText(message);
-                }
-            });
+            return doSend(SEND_TEXT_STR, message);
         }
 
         @Override
         public <M> Uni<Void> sendText(M message) {
-            return doSend(new Function<WebSocketConnection, Uni<Void>>() {
-
-                @Override
-                public Uni<Void> apply(WebSocketConnection c) {
-                    return c.sendText(message);
-                }
-            });
+            return doSend(SEND_TEXT_POJO, message);
         }
 
         @Override
         public Uni<Void> sendBinary(Buffer message) {
-            return doSend(new Function<WebSocketConnection, Uni<Void>>() {
+            return doSend(SEND_BINARY, message);
+        }
 
-                @Override
-                public Uni<Void> apply(WebSocketConnection c) {
-                    return c.sendBinary(message);
-                }
-            });
+        @Override
+        public Uni<Void> sendPing(Buffer data) {
+            throw new UnsupportedOperationException();
```

Comment:
You should probably return a failed `Uni` instead of throwing an exception?

Reply:
I'm not so sure. It's not a failure when sending a ping but an illegal operation and an exception thrown from a callback results in a failed `Uni` anyway (there is a generated try-catch block in the endpoint).

Reply:
Non-blocking code should in general never throw, but if there's a `catch` that turns an exception into a failed `Uni` in the generated code that calls this method, then it's fine.